### DISCUSSION TITLE
What would these FEEL expressions evaluate to?

### DISCUSSION THREAD

**Discussion by deostroll** [dmn-tck/tck#451] [Q&A] [answered]
```
> { a: 10, b: a + 10 }
> { b: a + 10, a: 12 }
```

I would want results per DMN 1.1. If it is by later version of spec please specify.

**Comment by StrayAlien** [comment, marked as answer]
Hi @deostroll ,

The first would be {a: 10, b:20}, the second would be {b: null, a:12}.

As context entries are evaluated, they are available in scope for subsequent entries.  For the second one, the `a` is not in scope so therefore would be `null`.  Adding `10` to `null` is `null`.

I'm not sure if that was for 1.1 or not (I came to DMN after 1.1), but that is what it is now. 

You'll find some tests here: https://github.com/dmn-tck/tck/tree/master/TestCases/compliance-level-3/0057-feel-context.

**Reply by tarilabs** [reply]
I concur with @StrayAlien , to be even more precise, the second one would be `{b: null, a:12}` ***and an error***.

DMNv1.1 onwards specify:
> An expression in a context entry (...) may reference keys (as QNs) from other context entries in the same context. These references SHALL be acyclic and form a partial order. The expressions in a context SHALL be evaluated consistent with this partial order.
